### PR TITLE
Feature/dropdown

### PR DESCRIPTION
**Pull request by ydomenjoud**
add dropdown
* breaking change * : kal-menu is now kal-nav

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu-trigger-for.directive.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      hasBackdrop: true,
+      minWidth: this.getHostWidth(),
+      scrollStrategy: this.overlay.scrollStrategies.reposition()
+    });
+
+    this.overlayRef.backdropClick().subscribe(() => {
+      this.closeMenu();
+    });
+
+    this.overlayRef.keydownEvents()
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu-trigger-for.directive.ts]: subscription.unsubscribe()

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu-trigger-for.directive.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    this.overlayRef = this.overlay.create({
+      positionStrategy: this.getPositionStrategy(),
+      panelClass: 'kal-menu__overlay',
+      hasBackdrop: true,
+      minWidth: this.getHostWidth(),
+      scrollStrategy: this.overlay.scrollStrategies.reposition()
+    });
+
+    this.overlayRef.backdropClick().subscribe(() => {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu-trigger-for.directive.ts]: subscription.unsubscribe()

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu-trigger-for.directive.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+   */
+  private openMenu(): void {
+    if (this.menuOpen) {
+      return;
+    }
+    const portal = new TemplatePortal(this.menu.templateRef, this.viewContainerRef);
+
+    this.getOverlay().attach(portal);
+
+    this.menu.closed.subscribe(() => {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu-trigger-for.directive.ts]: subscription.unsubscribe()

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu.component.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  resetKeyManager() {
+    this.keyManager.setActiveItem(-1);
+  }
+
+  ngAfterContentInit() {
+    this.keyManager = new ActiveDescendantKeyManager<KalOptionComponent>(this.options)
+      .withWrap()
+      .withVerticalOrientation()
+      .withTypeAhead();
+    this.tabSubscription = this.keyManager.tabOut.subscribe(() => this.closed.emit('tab'));
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu.component.ts]: you can use subscriptionsList

**Review by MichaelGrosio — CHANGES_REQUESTED** [review]

**Review by Bellico — COMMENTED** [review]
No readme and tests ?

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu.component.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  @ContentChildren(KalOptionComponent) options: QueryList<KalOptionComponent>;
+
+  @ViewChild(TemplateRef) templateRef: TemplateRef<any>;
+
+
+  /** Event emitted when the menu is closed. */
+  @Output() readonly closed: EventEmitter<void | 'click' | 'keydown' | 'tab' | 'pick'> =
+    new EventEmitter<void | 'click' | 'keydown' | 'tab' | 'pick'>();
+
+  @Output() readonly selectionChange = new  EventEmitter<KalOptionComponent>();
```
Comment (Bellico) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu.component.ts]: You emit all KalOptionComponent ? KalOption can have a value

Comment (ydomenjoud) [reply projects/kalidea/kaligraphi/src/lib/atoms/kal-menu/kal-menu.component.ts]: to get all information

**Review by MichaelGrosio — APPROVED** [review]

**Pull request merged**